src/mirage_net.mli

```diff
@@ -42,42 +42,47 @@ type stats = {
 module type S = sig
 
   type error = private [> Mirage_device.error]
-  (** The type for network errors. *)
+  (** The type for network interface errors. *)
 
   val pp_error: error Fmt.t
   (** [pp_error] is the pretty-printer for errors. *)
 
-  type page_aligned_buffer
-  (** The type for page-aligned memory buffers. *)
-
   type buffer
   (** The type for memory buffers. *)
 
   type macaddr
-  (** The type for unique MAC identifiers for the device. *)
+  (** The type for unique MAC identifiers for the network interface. *)
 
   include Mirage_device.S
 
   val write: t -> buffer -> (unit, error) result io
-  (** [write nf buf] outputs [buf] to netfront [nf]. *)
+  (** [write net buf] outputs [buf] to network interface [net]. *)
 
   val writev: t -> buffer list -> (unit, error) result io
-  (** [writev nf bufs] output a list of buffers to netfront [nf] as a
-      single packet. *)
+  (** [writev net bufs] output a list of buffers to the network interface [net]
+      as a single packet. *)
 
   val listen: t -> (buffer -> unit io) -> (unit, error) result io
-  (** [listen nf fn] is a blocking operation that calls [fn buf] with
+  (** [listen net fn] is a blocking operation that calls [fn buf] with
       every packet that is read from the interface. The function can
       be stopped by calling [disconnect] in the device layer. *)
 
+  val allocate_frame : t -> buffer
+  (** [allocate_frame net] allocates a single frame which can be filled
+      and transmitted via [write]. The network interface implementation knows
+      best about its memory-alignment requirements. *)
+
```

Comment:
Lgtm. Perhaps add a note here at that mixing frames allocated between different network devices is currently undefined, so dont do it?

Reply:
yes, I'll do that.